src/model.js

```diff
@@ -0,0 +1,226 @@
+  Object.defineProperty(object, field, {
+    get() {
+      const { isArray, resource } = this.constructor.attributes[field];
+      const model = Registry.get(resource);
+      if (!model) {
+        throw new Error(`Unregistered model: ${resource}`);
+      }
+      const data = this.data.relationships[field].data;
+
+      return isArray
+        ? Store.where({ id: data.map((d) => d.id) }, { model })
+        : Store.find({ id: data.id }, { model });
+    }
+  });
+};
+
+/**
+ * Define JSON-API attributes and relationships
+ * @ignore
+ */
+const defineMethods = (object, attributes) => {
+  const relationships = {};
+
+  Object.entries(attributes).forEach(([field, attribute]) => {
+    switch (attribute.type) {
+      case 'attribute':
+        defineAttribute(object.prototype, field, attribute);
+        break;
+      case 'relationship':
+        defineRelationship(object.prototype, field);
+        relationships[attribute.resource] = { key: field, ...attribute };
+        break;
+      default:
+        throw invalidAttribute(field);
+    }
+  });
+
+  return relationships;
+};
+
+/**
+ * @ignore
+ */
+const copyClassName = (from, to) => {
+  Object.defineProperty(to, 'name', {
+    value: from.name,
+    writable: false,
+    enumerable: false,
+    configurable: false,
+  });
+};
+
+/**
+ * Helper function to create endpoint URIs
+ * @ignore
+ * @example
+ * const obj = { type: 'animal', id: 5 }
+ * parseParams(obj, '/api/v1/:type/:id')
+ * // => '/api/v1/animal/5'
+ */
+export const parseParams = (context, string) => {
+  const regexp = /:[^/.]+/g;
+  return string.replace(regexp, (match) => context[match.substring(1)]);
+};
+
+/**
+  * Defines a model
+  * @alias module:active-redux.define
+  * @example
+  * const Person = ActiveRedux.define('people', class Person {});
+  * @param {string} type - JSON-API type for the model
+  * @param {Class} [model] - Class to extend
+  * @return {Model}
+  */
+export default function define(type, model = class {}) {
+  /**
+   * Active Redux Model
+   */
+  const Model = class extends model {
+
+    /**
+    * JSON-API type of the model
+    */
+    static type = type;
+
+    /**
+    * Attributes of a model, for which methods will be defined
+    * @example
+    * import Model, { Attr } from 'active-redux';
+    *
+    * const Person = Model.define('people', class Person {
+    *   static attributes = {
+    *     name: Attr.string(),
+    *   }
+    * });
+    *
+    * const joe = new Person({ attributes: { name: "Joe" } });
+    * joe.name // => "Joe"
+    */
+    static attributes = model.attributes || {};
+
+    /**
+    * Endpoints that will be queried. Can be overwritten.
+    */
+    static endpoints = model.endpoints || {
+      create: ':type',
+      read: ':type',
+      update: ':type/:id',
+      delete: ':type/:id',
+    };
+
+    /**
+    * @ignore
+    */
+    static relationships = {};
+
+    /**
+    * @ignore
+    */
+    static store = Store;
```

Comment:
Not sure we actually need to hang onto the store for anything - could refer to `Store` instead of `this.store` in methods below